### PR TITLE
feat(geo): merge Geo category into `main` branch

### PR DESCRIPTION
**Pull request by TreTuna**
#### Description of changes

- Adds `geo` category
- Updates `aws-amplify/core` with the config for Geo
- Updates `aws-amplify` to pull Geo through

#### Description of how you validated changes
- `@aws-amplify/geo@geo` developer preview version from this branch is released and used for testing
- Adds units tests
- Manually tested through demo applications

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sammartinez** [comment]
Do we have a reference on the bundle size increase on this?

**Comment by TreTuna** [comment]
> Do we have a reference on the bundle size increase on this?

From running the manual bundlesize checks, it looks like `FAIL dist/Amplify+Auth+Storage.js.min.js: 117.4KB > 115KB (gzip)` has gone up just over the threshold.

image
image

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
 		"publish:release": "lerna publish --conventional-commits --yes --message 'chore(release): Publish [ci skip]' --no-verify-access",
 		"publish:1.0-stable": "lerna publish --conventional-commits --yes --dist-tag=stable-1.0 --message 'chore(release): Publish [ci skip]' --no-verify-access",
 		"publish:ui-components/main": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=ui-preview --preid=ui-preview --exact --no-verify-access",
+		"publish:geo/main": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=geo --preid=geo --exact --no-verify-access",
```
Comment (sammartinez) [inline package.json:28]: Is this needed?

File: packages/geo/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
+# 0.0.2-geo.6654 (2021-09-13)
```
Comment (sammartinez) [inline packages/geo/CHANGELOG.md:6]: Your package start is set to `0.0.1` this seems to state `0.0.2` can we make sure which one is correct?

File: packages/geo/package.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
+{
+	"name": "@aws-amplify/geo",
+	"version": "0.0.1",
```
Comment (sammartinez) [inline packages/geo/package.json:3]: Same callout as above

File: package.json (hunk 1 above)
Comment (TreTuna) [reply package.json:28]: I'd prefer to leave this until after GA release at the moment. I have it listed as a fast-follow remove.

File: packages/geo/CHANGELOG.md (hunk 2 above)
Comment (TreTuna) [reply packages/geo/CHANGELOG.md:6]: Yeah, this seems to be an artifact of npm publish ticking the version up when it releases the tag, yet that version update is never committed. The breaking change did come with that tagged version though, so this is technically correct and when we do a full release tomorrow from `main` it should update this version.

File: packages/geo/package.json (hunk 3 above)
Comment (TreTuna) [reply packages/geo/package.json:3]: Same with rely from the others

**Review by iartemiev — APPROVED** [review]
LGTM 👍🏻

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Pull request merged**